### PR TITLE
[Feature] sanitizer on the SVG image

### LINKED ISSUE #13: [Feature] add fl_sanitize to support sanitizing on SVG image
# **Feature Request**

## **Is your feature request related to a problem? Please describe.**
Cloudinary has the fl_sanitize flag to support sanitizing on SVG image

link to feature request https://github.com/colbyfayock/next-cloudinary/issues/92

https://cloudinary.com/documentation/transformation_reference#fl_sanitize

## **Describe the solution you'd like**
Add `sanitizePlugin` to support SVG image

### PR DESCRIPTION
**Pull request by kitravee**
# Description

## Issue Ticket Number

Fixes #13 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Fix or improve the documentation
- [ ] This change requires a documentation update

# Checklist

- [x] I have followed the contributing guidelines of this project as mentioned in [CONTRIBUTING.md](/CONTRIBUTING.md)
- [x] I have created an [issue](https://github.com/colbyfayock/cloudinary-util/issues) ticket for this PR
- [x] I have checked to ensure there aren't other open [Pull Requests](https://github.com/colbyfayock/cloudinary-util/pulls) for the same update/change?
- [x] I have performed a self-review of my own code
- [x] I have run tests locally to ensure they all pass
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes needed to the documentation

### CODE REVIEW

**Comment by colbyfayock** [comment]
@kitravee this is great, thank you!

originally i was thinking to default the sanitize option to `true` - my thinking was that it should be beneficial to anyone and can help avoid issues. a "safer" default. though not 100% sure of the implications

what do you think? not opposed to leaving it "as is" but i like the idea of making it easier to be secure

**Comment by kitravee** [comment]
@colbyfayock  I agree with you 😁

I implement the plugin and set sanitize option to `true`
if there is other case that I am missing please comment, thank you colbyfayock

File: packages/url-loader/src/plugins/sanitize.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+export const props = ['sanitize'];
+
+export function plugin(props: PluginSettings) {
+  const { cldImage, options } = props;
+  const { sanitize = true } = options;
+
+  const shouldApplySanitizer: boolean =
+    sanitize &&
+    (options.format === 'svg' || cldImage.publicID.endsWith('.svg'));
```
Comment (colbyfayock) [inline packages/url-loader/src/plugins/sanitize.ts:11]: 🙌 nice add with the endswith. totally would have forgot this one ha

**Comment by colbyfayock** [comment]
hey @kitravee made 2 changes on here:

* in the plugin code, you were returning the image, which was i think making things work a little weird, not totally sure, but just removed the return statement

* I updated the plugin test code to work like the other plugin tests. I import ONLY The plugin and run tests based on only the plugin code. so i used the exact same settings and tests you had in there already, only i changed the format of how they ran

this looks great though! thanks for knocking this out

**Comment by colbyfayock** [comment]
@all-contributors please add @kitravee for code

**Comment by colbyfayock** [comment]
@all-contributors please add @kitravee for code

**Pull request merged**